pkg/raft/log.go

```diff
@@ -531,6 +550,27 @@ func (l *raftLog) scan(lo, hi uint64, pageSize entryEncodingSize, v func([]pb.En
 // The returned slice can be appended to, but the entries in it must not be
 // mutated.
 func (l *raftLog) slice(lo, hi uint64, maxSize entryEncodingSize) ([]pb.Entry, error) {
+	return l.snap().slice(lo, hi, maxSize)
+}
+
+// LogSlice returns a valid log slice for the (lo, hi] log index interval.
+func (l LogSnapshot) LogSlice(lo, hi uint64, maxSize uint64) logSlice {
+	prevTerm, err := l.term(lo)
+	if err != nil { // the log is probably compacted at index > lo
+		return logSlice{}
+	}
+	ents, err := l.slice(lo, hi, entryEncodingSize(maxSize))
+	if err != nil { // the log is compacted
```

Comment:
Not necessarily, could also be an SST checksum error 🤷🏽

Reply:
Removed this comment.